src/compas/datastructures/tree/tree.py

```diff
@@ -450,22 +449,27 @@ def print_hierarchy(self, max_depth=None):
 
         Returns
         -------
-        None
+        str
+            The spatial hierarchy of the tree.
```

Comment:
fwiw, i don't agree with this. why would anyone expect `Tree.hierarchy` to return a string?

Reply:
if the only purpose of this method is to be a helper for `__str__`, i think it should be named differently and made private.

Reply:
How about something more expressive like `get_hierarchy_string` ? Tho I'm more on the side for still keeping it public because I call this function a lot with different `max_depth` for very deep trees, I can imagine people might also want to do something like this

Reply:
@tomvanmele any comments on this? would like to resolve it